core/commands/fixup.py

```diff
@@ -59,6 +59,7 @@ def do_action(self, commit, **kwargs):
 
             self.rewrite_active_branch(base_commit, commit_chain)
         except Exception as e:
+            self.git("reset", "--hard", commit)
```

Comment:
Why do we want to do this?
If we do a reset --hard commit (This is the commit you want to rebase the changes into??), would we remove the commits from commit unto HEAD?

Reply:
Ai, we should reset only....reset —hard would be too much. The idea is to get rid of the fixup commit.

Reply:
`HEAD^1` ?
So one commit before HEAD?

Reply:
👍